File: solutions/1773. Count Items Matching a Rule.cpp
class Solution {
public:
    int countMatches(vector<vector<string>>& items, string ruleKey, string ruleValue) {
        int rk = 0;
        if (ruleKey == "color") rk = 1;
        else if (ruleKey == "name") rk = 2;
        int count = 0;
        for (auto& strVec : items) {
            if (strVec.at(rk) == ruleValue) ++count;
        }
        return count;
    }
};

File: solutions/1773. Count Items Matching a Rule.rs
impl Solution {
    pub fn count_matches(items: Vec<Vec<String>>, rule_key: String, rule_value: String) -> i32 {
        let index = match rule_key.as_str() {
            "type" => 0,
            "color" => 1,
            "name" => 2,
            _ => panic!(),
        } as usize;
        items.iter().fold(0, |acc, item| {
            acc + if item[index] == rule_value { 1 } else { 0 }
        })
    }
}
